pkg/vault/test/foundry/fork/YieldBearingPoolSwapBase.t.sol

```diff
@@ -1024,96 +929,94 @@ abstract contract YieldBearingPoolSwapBase is BaseVaultTest {
         } else {
             bytes32 bufferBalances = vault.getBufferTokenBalancesBytes(wToken);
             // Deposit converts underlying to wrapped. If buffer has a surplus of underlying, the vault wraps it to
             // rebalance the buffer. It can introduce rounding issues, so we should consider the rebalance in our result
             // preview.
-            uint256 underlyingSurplus = bufferBalances.getBufferUnderlyingSurplus(wToken);
+            underlyingSurplus = bufferBalances.getBufferUnderlyingSurplus(wToken);
 
             // Do the actual operation to impact the rate used to calculate wrapped surplus.
             uint256 snapshotId = vm.snapshot();
             vm.startPrank(lp);
             IERC20(wToken.asset()).forceApprove(address(wToken), underlyingToDeposit + underlyingSurplus);
             uint256 vaultWrappedDelta = wToken.deposit(underlyingToDeposit + underlyingSurplus, lp);
             vm.stopPrank();
-            uint256 wrappedSurplusToLeaveInTheBuffer = wToken.convertToShares(underlyingSurplus);
+            wrappedSurplus = wToken.convertToShares(underlyingSurplus);
             vm.revertTo(snapshotId);
 
-            return vaultWrappedDelta - wrappedSurplusToLeaveInTheBuffer;
+            mintedShares = vaultWrappedDelta - wrappedSurplus;
         }
     }
 
     function _previewMintWithBuffer(
         IERC4626 wToken,
         uint256 wrappedToMint,
         bool shouldUseConvert
-    ) private view returns (uint256) {
+    ) private view returns (uint256 depositedAssets, uint256 underlyingSurplus, uint256 wrappedSurplus) {
         if (shouldUseConvert) {
             // If operation is within buffer range, convert is used by the vault to save some gas.
-            return wToken.convertToAssets(wrappedToMint);
+            depositedAssets = wToken.convertToAssets(wrappedToMint) + vaultConvertFactor;
         } else {
             bytes32 bufferBalances = vault.getBufferTokenBalancesBytes(wToken);
             // Mint converts underlying to wrapped. If buffer has a surplus of underlying, the vault wraps it to
             // rebalance the buffer. It can introduce rounding issues, so we should consider the rebalance in our
             // result preview.
-            uint256 underlyingSurplus = bufferBalances.getBufferUnderlyingSurplus(wToken);
-            uint256 wrappedSurplusToLeaveInTheBuffer = wToken.convertToShares(underlyingSurplus);
-            uint256 vaultUnderlyingDelta = wToken.previewMint(wrappedToMint + wrappedSurplusToLeaveInTheBuffer);
-            return vaultUnderlyingDelta - underlyingSurplus;
+            underlyingSurplus = bufferBalances.getBufferUnderlyingSurplus(wToken);
+            wrappedSurplus = wToken.convertToShares(underlyingSurplus);
+            uint256 vaultUnderlyingDelta = wToken.previewMint(wrappedToMint + wrappedSurplus);
+            depositedAssets = vaultUnderlyingDelta - underlyingSurplus;
```

Comment:
This bit is replicating the logic in the vault right?

Reply:
Yes, I added some comments to explain this better.